lib/pages/search/server_search_controller.dart

```diff
@@ -71,19 +74,24 @@ class ServerSearchController with SearchDebouncerMixin {
       (success) {
         if (success is ServerSearchChatSuccess) {
           updateNextBatch(success.nextBatch);
-          searchResultsNotifier.value = PresentationServerSideSearch(
-            searchResults: [
-              ...searchResultsNotifier.value.searchResults,
-              ...success.results ?? [],
-            ],
-          );
+          if (success.results?.isEmpty == true) {
+            searchResultsNotifier.value = PresentationServerSideEmptySearch();
+          } else {
+            searchResultsNotifier.value = PresentationServerSideSearch(
+              searchResults: [
+                ...(searchResultsNotifier.value as PresentationServerSideSearch)
+                    .searchResults,
+                ...success.results ?? [],
+              ],
+            );
+          }
         }
       },
     );
   }
 
   void resetSearchResults() {
-    searchResultsNotifier.value = const PresentationServerSideSearch(
+    searchResultsNotifier.value = PresentationServerSideSearch(
       searchResults: [],
     );
```

Comment:
why not use `PresentationServerSideEmptySearch`, please use 1 class, make it less confused

Reply:
It's func reset search result, not case empty